apps/gallery/js/ImageEditor.js

```diff
@@ -229,134 +405,20 @@ function editAspectMoreHandler() {
-    var delta = navigator.mozL10n.language.direction === 'ltr' ?
-                e.detail.absolute.dx : - e.detail.absolute.dx;
-
-    var exposureDelta = delta / parseInt(bar.clientWidth, 10) * 6;
-    // For the firt time of pan event triggered
-    // set start value to current value.
-    if (!dragStart)
-      dragStart = currentExposure;
-
-    setExposure(dragStart + exposureDelta);
-    e.preventDefault();
-  });
-  thumb.addEventListener('swipe', function(e) {
-    // when stopping we init the dragStart to be null
-    // this way we avoid a 'panstart' event
-    dragStart = null;
-    e.preventDefault();
-  });
-  thumb.addEventListener('touchstart', function(e) {
-    thumb.classList.add('active');
-  });
-  thumb.addEventListener('touchend', function(e) {
-    thumb.classList.remove('active');
-  });
-  slider.addEventListener('keypress', function(e) {
-    // screen reader sends key arrow up/down events for adjusting the slider.
-    if (e.keyCode == KeyEvent.DOM_VK_DOWN) {
-      setExposure(currentExposure - 0.25);
-    } else if (e.keyCode == KeyEvent.DOM_VK_UP) {
-      setExposure(currentExposure + 0.25);
-    }
-  });
-
-  function resize() {
-    forceSetExposure(currentExposure);
-  }
-
-  // Set the thumb position between -3 and +3
-  function setExposure(exposure) {
-    // Make sure it is not out of bounds
-    if (exposure < -3)
-      exposure = -3;
-    else if (exposure > 3)
-      exposure = 3;
-
-    // Round to the closest sixteenth
-    exposure = Math.round(exposure * 16) / 16;
-
-    if (exposure === currentExposure)
-      return;
-
-    forceSetExposure(exposure);
-  }
-
-  function forceSetExposure(exposure) {
-    // Remove left/right 4 pixels of margin
-    var barWidth = parseInt(bar.clientWidth, 10) - 4 * 2;
-    var thumbWidth = parseInt(thumb.clientWidth, 10);
-    var offset = bar.offsetLeft + 4;
-
-    // Convert exposure value to a unit coefficient position of thumb center
-    var unitCoef = (exposure + 3) / 6;
-
-    // Convert unitCoef to pixel position of thumb center
-    var pixel = offset + barWidth * unitCoef;
-
-    // Compute pixel position of left edge of thumb
-    pixel -= thumbWidth / 2;
-
-    // Move the thumb to that position
-    thumb.style.MozMarginStart = pixel + 'px';
-
-    // Display exposure value in thumb
-    thumb.textContent = exposure;
-
-    // Don't need to update the currentExposure and dispatch the event if they
-    // are the same.
-    if (currentExposure === exposure) {
-      return;
-    }
-    // Remember the new exposure value
-    currentExposure = exposure;
-
-    // Set value for ARIA widget
-    slider.setAttribute('aria-valuenow', exposure);
-
-    // Dispatch an event to actually change the image
-    slider.dispatchEvent(new Event('change', {bubbles: true}));
-  }
-
-  return {
-    resize: resize,
-    forceSetExposure: forceSetExposure,
-    setExposure: setExposure,
-    getExposure: function() { return currentExposure; }
-  };
-})();
-
```

Comment:
nit: If JSHint allows below slider on change handler line 408 - 436 should be moved above with the exposureSlider code at Line 221